cs_api/cs_api/api/runs.py

```diff
@@ -88,3 +96,90 @@ def get_runs_from_interests():
                 f"filter_by parameter must be either 'sources' or 'sites', not {filter_by}"
             )
     return flask.jsonify(filtered_runs)
+
+
+@server.app.route(const.ADD_RUN, methods=["POST"])
+@cross_origin(expose_headers=["Content-Type", "Authorization"])
+@utils.endpoint_exception_handling(server.app)
+def add_run():
+    """
+    Adds the run to the database
+    """
+    server.app.logger.info(f"Received request at {const.ADD_RUN}")
+    (
+        run_name,
+        dropbox_folder,
+        run_type,
+        region,
+        tectonic_types,
+        grid_spacing,
+        secret_key,
+    ) = utils.get_check_keys(
+        flask.request.args,
+        (
+            "run_name",
+            "dropbox_folder",
+            "run_type",
+            "region",
+            "tectonic_types",
+            "grid_spacing",
+            "secret_key",
+        ),
+    )
+    tectonic_types = tectonic_types.split(",")
+
+    # Check the secret key is correct
+    if secret_key != os.environ["SECRET_KEY"]:
+        return flask.jsonify({"error": "Incorrect secret key"}), 401
+    else:
+        dropbox_directory = f'dropbox:{"/".join(dropbox_folder.split("/")[2:])}'
+
+        # Load the site_df by taking the current files path and navigating to the resouces directory
+        site_df_ffp = Path(__file__).parent.parent / "db" / "resources" / "site_df.csv"
+        site_df = pd.read_csv(site_df_ffp, index_col=0)
+        # Create a new column for the current run and set all to False
+        site_df[run_name] = False
+
+        # Create a temporary directory to get the site information for the run
+        temp_dir = TemporaryDirectory()
+        temp_path = Path(temp_dir.name).resolve()
+
+        cs_dropbox_download_py_ffp = os.environ["CS_DROPBOX_DOWNLOAD_PY_FFP"]
+
+        # Download the IM data
+        p = subprocess.Popen(
```

Comment:
Any particular reason for calling python in a subprocess and not just calling it directly?

Reply:
Very fair point, forgot I made a function to do this easier